### DISCUSSION TITLE
Is there a way to specify Bible translation on verse entry?

### DISCUSSION THREAD

**Discussion by jamesx4** [tim-hub/obsidian-bible-reference#204] [Q&A]
Is there some way to specify a Bible translation to use other than the default when entering a verse?  Sometimes I'd like to place verses from different translations side-by-side.  I didn't see anything in the help docs.  If not available, that would make a great feature!

Great work!  One of the most useful plugins for me!

**Comment by VIRGO-AIORIA** [comment]
Yes. For now there's no alternative other than going to the plugin settings each time one would want to. It'd be great.
I reckon it in its minimum deliverable way would be to change the link. For example,

Having NKJV as Default means that the output of `Hebrews 4:7` will be  
>    7 again He designates a certain day, saying in David, “Today,” after such a long time, as it has been said: “Today, if you will hear His voice, Do not harden your hearts.”
https://bolls.life/NKJV/58/4/#7

However, a different translation that is also in the same website would be mostly changing the link to the translation's, no? in Spanish, `Hebrews 4:7 RV1960` takes us to
https://bolls.life/RV1960/58/4/#7
Still the same site but changes just book portion, and both work.

Same with Bible API, which changes the book suffix at the end. 
`Hebreos 4:12`
https://www.biblegateway.com/passage/?search=Hebreos+4%3A12&version=RVR1960
is distinct from
https://www.biblegateway.com/passage/?search=Hebreos+4%3A12&version=nkjv
and both work!

It'd a great addition and possibly what's needed for this to be best Bible plugin in Obsidian I've seen so far. Thank you very much!

**Reply by ctbarber** [reply]
I think this is available already. Typing
`--John3:16-ESV` will switch to the ESV version but leave the default version intact.

EDIT: Sorry @Kilarin and @tim-hub for the typo. I meant to have the 17 in there. Proper format is in a [comment below](https://github.com/tim-hub/obsidian-bible-reference/discussions/204#discussioncomment-17928367). When I used it, I didn't include any spaces after the last verse and it did work.

**Reply by Kilarin** [reply]
This does NOT work for me.

**Reply by tim-hub** [reply]
> This does NOT work for me.

This supposed to work. I will have a look why.

**Reply by Kilarin** [reply]
Interesting.  well then, let me give you some more information.
I'm on Obsidian version 1.13.4 running on Linux Ubuntu 24.04.4
Bible Reference version 26.07.29-1
I can type:
--John3:16-ESV
or
--john3:16-17-KJV
and I get nothing. 
When I type: --john3:16-17   The verse pops up
<img width="523" height="152" alt="image" src="https://github.com/user-attachments/assets/3ca48870-4751-464e-9d79-74537d4dcbf4" />

But as soon as I type the - it disappears, and does not come back when I complete the version abbreviation.
<img width="297" height="79" alt="image" src="https://github.com/user-attachments/assets/30851ea0-e441-4251-b5cb-dcfb24a37f70" />

Hitting enter does not populate it either.

My settings are:
<img width="598" height="1127" alt="image" src="https://github.com/user-attachments/assets/89a3ca95-0d94-4cf4-81f6-e6f2a46151b5" />
<img width="599" height="1132" alt="image" src="https://github.com/user-attachments/assets/22b8cce5-96e1-426b-b667-6e471052b7a3" />

And again, thank you so very much!  This is an incredibly valuable tool.  The time and effort you have put into it is very much appreciated.

**Reply by tim-hub** [reply]
try this
`@` is the separator for the version:

```
--John 3:16@esv
--john3:16-17@kjv
```

**Reply by Kilarin** [reply]
This is PERFECT!  Thank you!

**Reply by Kilarin** [reply]
One question on this.  When I use the @ sign, it now changes my default preference.  Is that by design?  
To me, it is intuitive to change my default preference in options, and to use the @ for a temporary override.
Not a big important issue, it's simple enough to change it back with another @.

**Reply by tim-hub** [reply]
Good catch — yes, that's the current behaviour, and your reading of what's intuitive matches mine.

Under the hood there are two values: your default version (the one you pick in settings) and the active version used for lookups. `@esv` only changes the active one — your stored default is untouched. It's sticky because the same active-version value backs the quick translation switcher hotkey, so a version you pick that way keeps applying to the next verses you type. `@` was built on top of that mechanism and inherited the stickiness.

One side effect: the version dropdown in settings shows the active version, so after `@esv` it reads ESV — which makes it look like your default changed when it hasn't.

To get back without typing another `@`:
- Command palette → **Bible Reference: Switch to Default Translation**

That said, I agree `@` reads more like a one-time override. I'll track making `@` apply to just that one verse while the hotkey stays sticky. Does that match what you'd expect?

**Reply by Kilarin** [reply]
Certainly does.  THANK YOU!
